scripts/mrbind-pybind11/install_pythons_macos.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Most of this wizardry is copied from our previous pip-build file, not sure why some of this is needed.
+
+set -euxo pipefail
+
+# Load the list of Python versions. `xargs` trims the whitespace and removes newlines.
+SCRIPT_DIR="$(realpath "$(dirname "$BASH_SOURCE")")"
+PY_VERSIONS="$(cat $SCRIPT_DIR/python_versions.txt | xargs)"
+
+# This seems to be the default location on Arm Macs, while x86 Macs use `/usr/local`.
+HOMEBREW_DIR=/opt/homebrew
+[[ -d $HOMEBREW_DIR ]] || HOMEBREW_DIR=/usr/local
+
+if [[ ${ENABLE_SUDO:=} == 1 ]]; then
+    SUDO=sudo
+elif [[ ${ENABLE_SUDO:=} == 0 ]]; then
+    SUDO=
+elif which sudo >/dev/null 2>/dev/null; then
+    SUDO=sudo
+else
+    SUDO=
+fi
+
+# ??
+$SUDO find "$HOMEBREW_DIR/bin" -lname '*/Library/Frameworks/Python.framework/*' -delete
+$SUDO rm -rf /Library/Frameworks/Python.framework/
```

Comment:
Very bad idea to remove anything from user's OS. If it's required for our CI, move the commands to the workflow files.